### ISSUE TITLE
Push Notifications Feature

### ISSUE THREAD

**Issue by Prashant2504** [developer-student-club-thapar/mobile-app#8]
found this package. 

https://pub.dev/packages/firebase_messaging

we can use this for our push notifiations

**Comment by ramyak-mehra** [comment]
Yup i was also thinking of using this we need the firebase keys and json file for this.

**Comment by desaihetav** [comment]
I've used OneSignal Notifications in one of my Flutter projects. Firebase Messaging didn't work consistently and I guess it has some issues with Flutter. I'd like to work on this issue.

**Comment by ramyak-mehra** [comment]
> I've used OneSignal Notifications in one of my Flutter projects. Firebase Messaging didn't work consistently and I guess it has some issues with Flutter. I'd like to work on this issue.

Last time I used notifications it worked what were the problems you faced?

**Comment by desaihetav** [comment]
> Last time I used notifications it worked what were the problems you faced?
> 

The notifications worked perfectly when the app was in foreground or in killed state. But when the app was in background and if I sent 3-4 notifications with a gap of 2-5 mins between them, it wouldn't work consistently i.e. I didn't receive all the notifications.

I tested quite extensively but it didn't seem to work and hence I shifted to OneSignal.

**Comment by ramyak-mehra** [comment]
> > Last time I used notifications it worked what were the problems you faced?
> > 
> 
> The notifications worked perfectly when the app was in foreground or in killed state. But when the app was in background and if I sent 3-4 notifications with a gap of 2-5 mins between them, it wouldn't work consistently i.e. I didn't receive all the notifications.
> 
> I tested quite extensively but it didn't seem to work and hence I shifted to OneSignal.

So basically one signal also uses firebase only because it's required by play store. We don't want to be dependent upon any third service other than firebase.

**Comment by desaihetav** [comment]
> > > Last time I used notifications it worked what were the problems you faced?
> > > 
> > 
> > The notifications worked perfectly when the app was in foreground or in killed state. But when the app was in background and if I sent 3-4 notifications with a gap of 2-5 mins between them, it wouldn't work consistently i.e. I didn't receive all the notifications.
> > 
> > I tested quite extensively but it didn't seem to work and hence I shifted to OneSignal.
> 
> So basically one signal also uses firebase only because it's required by play store. We don't want to be dependent upon any third service other than firebase. 

Yup, but I guess that's for Android only.

**Comment by ramyak-mehra** [comment]
> > > > Last time I used notifications it worked what were the problems you faced?
> > > > 
> > > 
> > > The notifications worked perfectly when the app was in foreground or in killed state. But when the app was in background and if I sent 3-4 notifications with a gap of 2-5 mins between them, it wouldn't work consistently i.e. I didn't receive all the notifications.
> > > 
> > > I tested quite extensively but it didn't seem to work and hence I shifted to OneSignal.
> > 
> > So basically one signal also uses firebase only because it's required by play store. We don't want to be dependent upon any third service other than firebase. 
> 
> Yup, but I guess that's for Android only.

We are currently focusing on android only

**Comment by bibekpanda55** [comment]
I want to work on this issue can you please assign it to me

**Comment by bibekpanda55** [comment]
> Yup i was also thinking of using this we need the firebase keys and json file for this.

Yes we also need a blaze plan for the push notifications using firebase messaging

**Comment by ramyak-mehra** [comment]
> I want to work on this issue can you please assign it to me

Sure go on .
Also if you haven't registered for hacktoberfest you can go here http://dsctiet.xyz/hacktoberfest

**Comment by bibekpanda55** [comment]
Yes I have already registered

**Comment by bibekpanda55** [comment]
You do have a blaze plan on firebase right otherwise the notification wont work?

**Comment by ramyak-mehra** [comment]
> You do have a blaze plan on firebase right otherwise the notification wont work?

I don't think so that's the case 
I have used notifications on spark plan

**Comment by bibekpanda55** [comment]
Ok I will look for that ,will notify you if there's an error.

**Comment by bibekpanda55** [comment]
If we use node 10 we require a blaze pan for push notification but for node 8 or previous versions we can stick to spark plan but that would be valid till 2020 only and from 2021 it would be mandatory to use node 10 with blaze plan for push  notification

**Comment by ramyak-mehra** [comment]
> If we use node 10 we require a blaze pan for push notification but for node 8 or previous versions we can stick to spark plan but that would be valid till 2020 only and from 2021 it would be mandatory to use node 10 with blaze plan for push notification

Ok I will look into it

**Comment by bibekpanda55** [comment]
Yeah sure

**Comment by bibekpanda55** [comment]
Any update?

**Comment by ramyak-mehra** [comment]
> Any update?

I guess we would be sticking to firebase only we have credits.

**Comment by bibekpanda55** [comment]
So should I go onto use Node 8 ,it is depreciated but will work till end of 2020?
